### DISCUSSION TITLE
Run Single Test onSave

### DISCUSSION THREAD

**Discussion by AlessandroVol23** [orta/vscode-jest#1069] [Q&A]
Hi 👋🏽
I really like the extension but I'm missing one setting (maybe I just don't find the correct setting).

Once I hit save I'd like to only run one single test instead of the whole file. My current setting for autorun is:

```json
"jest.autoRun": {
   "watch": false,
   "onSave": "test-file"
}
```

Any idea how I can get that running? Or is that not possible right now? 

Thanks 😊

**Comment by connectdotz** [comment]
hi, the current implementation for on-save is to run the whole file. 

I assume you were hoping to run only the tests that changed? It's an interesting idea, and I can see that it might be useful for the files with lots of tests... 🤔

**Reply by AlessandroVol23** [reply]
Hi 👋🏽
yes exactly I was hoping that I can only run a single test while saving.
Thanks for the reply :)
